### PR TITLE
Set client timeout https://github.com/Kong/deck/issues/449

### PR DESCRIPTION
**Pull request by fjaenni**
#449

### CODE REVIEW

File: utils/types.go [hunk 1]
```diff
@@ -23,6 +23,8 @@ const (
 	defaultClientTimeout = 10 * time.Second
 )
 
+var ClientTimeout  time.Duration
```
Comment (mmorel-35) [inline utils/types.go:26]:
```suggestion
var clientTimeout  time.Duration
```

File: utils/types.go [hunk 2]
```diff
@@ -129,6 +133,11 @@ func GetKongClient(opt KongClientConfig) (*kong.Client, error) {
 		}
 		tlsConfig.RootCAs = certPool
 	}
+	
+	ClientTimeout = defaultClientTimeout
```
Comment (mmorel-35) [inline utils/types.go:137]:
```suggestion
	clientTimeout = defaultClientTimeout
```

File: utils/types.go [hunk 3]
```diff
@@ -129,6 +133,11 @@ func GetKongClient(opt KongClientConfig) (*kong.Client, error) {
...
+	if opt.Timeout > 0{
+		ClientTimeout =  time.Duration(opt.Timeout) * time.Second
```
Comment (mmorel-35) [inline utils/types.go:139]:
```suggestion
		clientTimeout =  time.Duration(opt.Timeout) * time.Second
```

File: utils/types.go [hunk 4]
```diff
@@ -204,12 +213,12 @@ func CleanAddress(address string) string {
 // sane default timeouts.
 func HTTPClient() *http.Client {
 	return &http.Client{
-		Timeout: defaultClientTimeout,
+		Timeout: ClientTimeout,
```
Comment (mmorel-35) [inline utils/types.go:216]:
```suggestion
		Timeout: clientTimeout,
```

File: utils/types.go [hunk 5]
```diff
@@ -204,12 +213,12 @@ func CleanAddress(address string) string {
...
 		Transport: &http.Transport{
 			DialContext: (&net.Dialer{
-				Timeout: defaultClientTimeout,
+				Timeout: ClientTimeout,
```
Comment (mmorel-35) [inline utils/types.go:219]:
```suggestion
				Timeout: clientTimeout,
```

File: utils/types.go [hunk 6]
```diff
@@ -204,12 +213,12 @@ func CleanAddress(address string) string {
...
 			}).DialContext,
-			TLSHandshakeTimeout: defaultClientTimeout,
+			TLSHandshakeTimeout: ClientTimeout,
```
Comment (mmorel-35) [inline utils/types.go:221]:
```suggestion
			TLSHandshakeTimeout: clientTimeout,
```

**Review by mmorel-35 — COMMENTED** [review]
Does it really needs to be a public variable? Lowercase shall be enough?

**Review by franciscojn — DISMISSED** [review]

**Review by rainest — CHANGES_REQUESTED** [review]
@fjaenni some minor linter issues that need addressing.

I'd recommend placing the default timeout in the flag (`rootCmd.PersistentFlags().Int("timeout", 10, ...)`) and using that value throughout. Otherwise it's a bit confusing that the value shown in help doesn't reflect the actual default timeout--it suggests that the default is instead no timeout.

**Comment by fjaenni** [comment]
> @fjaenni some minor linter issues that need addressing.
> 
> I'd recommend placing the default timeout in the flag (`rootCmd.PersistentFlags().Int("timeout", 10, ...)`) and using that value throughout. Otherwise it's a bit confusing that the value shown in help doesn't reflect the actual default timeout--it suggests that the default is instead no timeout.

hi,
I agree, I thought about doing it like this but finally ...
Thank you so much

**Review by rainest — APPROVED** [review]

**Pull request merged**